packages/ingest/granule.js

```diff
@@ -85,25 +89,39 @@ class Discover {
     // select files that match a given collection
     files.forEach(f => {
       const file = this.setGranuleInfo(f);
+      //console.log(file);
       if (file) updatedFiles.push(file);
     });
+    //console.log(updatedFiles);    
     return await this.findNewGranules(updatedFiles);
   }
 
   async fileIsNew(file) {
+  // If the bucket is defined 
+  if(file.bucket){
```

Comment:
@amarouane-ABDLHAK what are the cases where the file is missing a bucket?

Reply:
It could happen if the granuleIdExtraction regex matches the provided file but it does not match any regex within collection.meta.files

Reply:
I thought files that are not matched in collection.meta.files are ignored, that's not the case?

Reply:
I thought that too but I got some undefined bucket :) and digging into the source code I saw that the files not matching the regex but matching the granuleIdExtraction got the undefined buckets.

Reply:
interesting. This is certainly something we have to fix.

Reply:
Filed a ticket for it: https://bugs.earthdata.nasa.gov/browse/CUMULUS-261